arviz/data/inference_data.py

```diff
@@ -324,6 +325,59 @@ def sel(
         else:
             return out
 
+    def add_groups(self, group_dict=None, coords=None, dims=None, **kwargs):
+        """Add new groups to InferenceData object.
+
+        Parameters
+        ----------
+        group_dict: dict[str] -> dict, xr.Dataset
```

Comment:
```suggestion
        group_dict: dict of {str : dict or xarray.Dataset}, optional
```

Reply:
The above is following the guidance on pandas docstring guide, however, pandas docs also use the pattern `dict of str -> dict or xarray.Dataset`. Here I kind of prefer the first alternative due to the or, in general both are fine.